btree.go

```diff
@@ -282,25 +279,28 @@ func (btree *btreeTable) parseCell() (cellInfo, error) {
 		// sz is the total payload size.
 		// check if all of it is in the b-tree leaf page or
 		// if it spilled over to other pages
-		localsz := int(sz)
-		U := btree.page.PageSize() - int(btree.db.header.NReserved)
-		M := int(((U - 12) * 32 / 255.) - 23)
 		P := int(sz)
-		if P > U-35 {
-			vv := M + ((P - M) % (U - 4))
-			localsz = min(vv, U-35)
+		U := btree.page.PageSize() - int(btree.db.header.NReserved)
+		X := U - 35
+		localsz := P
+		overflowsz := 0
+		if P > X {
+			M := int(((U - 12) * 32 / 255) - 23)
+			K := M + ((P - M) % (U - 4))
+			localsz = K
+			if K > X {
+				localsz = M
+			}
+			overflowsz = P - localsz
 		}
 
 		// FIXME(sbinet): only create a new payload []byte when non-local
 		// ie: when there is an overflow page
 		payload := make([]byte, localsz, localsz)
-		n, err := btree.page.Read(payload)
+		_, err := btree.page.Read(payload)
```

Comment:
we could perhaps make that clear by using `io.ReadFull` ?
(I had a few local modifications doing just that but you seem to be on a good trend :))

Reply:
done